Makefile

```diff
@@ -312,14 +313,19 @@ $(BOOT_BUILD)/boot.bin: $(BOOT_BUILD)/boot.bif $(FSBL_ELF) $(U_BOOT_ELF)
 $(BOOT_ZIP): $(BOOT_FILES)
 	mkdir -p $(BOOT_IMAGE)
 	cp $^ $(BOOT_IMAGE)
-	zip -j $(BOOT_ZIP) $^
+	zip -j $@ $^
 
-boot: $(BOOT_ZIP)
+SRC_DIRS = $(wildcard $(SRC_ROOT)/*)
+SRC_TARS = $(wildcard $(patsubst $(SRC_ROOT)/%,$(TAR_FILES)/%*,$(SRC_DIRS)))
+$(BOOT_DEPS): $(BOOT_ZIP)
+	zip -j $@ $(SRC_TARS)
+
+boot: $(BOOT_ZIP) $(BOOT_DEPS)
```

Comment:
Not really right for `BOOT_ZIP` and `BOOT_DEPS` to be dependencies of `boot` target, probably best to set a separate target for these.